spec/codelog/command/regenerate_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+describe Codelog::Command::Regenerate do
+  describe '.run' do
+    it 'calls the "Changelog" step to re-generate the CHANGELOG using existing releases' do
+      allow(Codelog::Command::Step::Changelog).to receive(:run)

```

Comment:
@emilianoLeite I was recently questioning to what extent we should mock the services in this kind of test. If, for some reason, `Codelog::Command::Step::Changelog#run` is changed to have some new arbitrary behavior its own test would fail but not this one, that might lead to this command having a completely different behavior with no one ever noticing it by the tests.
What are yours views on that matter? Do you think high level feature tests are worth to prevent this kind of problems?

Reply:
@luisbevilacqua I think that's the idea, if the service is being reused here, does not make to retest it on this case (the command does only the service call, and nothing else).

Reply:
@luisbevilacqua For integration tests such as this, I believe it is best to mock collaborating classes, but I also agree that the application is brittle without more integrated tests.

I believe the best course of action would be to try to write some e2e tests, where we wouldn't mock any classes.